tests.c

```diff
@@ -1176,6 +1176,62 @@ test_opcode_0xfe(void) // NOLINT
   cpu_write_mem(&cpu, 0x0005, 0x00); // NOLINT
 }
 
+void
+test_handle_interrupt(void) // NOLINT
+{
+  i8080 cpu;
+  cpu_init(&cpu);
+  cpu.interrupt_enabled = true;
+  cpu.pc = 0xAABB; // NOLINT
+  cpu.sp = 0xFFFF; // NOLINT
+
+  int interrupt_handled = handle_interrupt(&cpu, 0x01); // NOLINT
+
+  CU_ASSERT(interrupt_handled == 0);
+  CU_ASSERT(cpu.pc == 0x0008);                       // NOLINT
+  CU_ASSERT(cpu.sp == 0xFFFD);                       // NOLINT
+  CU_ASSERT(cpu.interrupt_enabled == false);         // NOLINT
+  CU_ASSERT(cpu_read_mem(&cpu, cpu.sp) == 0xBB);     // NOLINT
+  CU_ASSERT(cpu_read_mem(&cpu, cpu.sp + 1) == 0xAA); // NOLINT
+
+  // clean up
+  cpu_write_mem(&cpu, 0xFFFE, 0x00); // NOLINT
+  cpu_write_mem(&cpu, 0xFFFD, 0x00); // NOLINT
+}
+
+void
+test_handle_interrupt_invalid_rst(void) // NOLINT
+{
+  i8080 cpu;
+  cpu_init(&cpu);
+  cpu.interrupt_enabled = true;
+  cpu.pc = 0xAABB; // NOLINT
+  cpu.sp = 0xFFFF; // NOLINT
+
+  int interrupt_handled = handle_interrupt(&cpu, 0x08); // NOLINT
+
+  CU_ASSERT(interrupt_handled == -1);
+  CU_ASSERT(cpu.interrupt_enabled == true);
```

Comment:
It's makes sense that this is persisted in the event of an invalid RST, but is this documented anywhere?

Reply:
I don't think the 8080 manual weighs in on whether interrupts are disabled after an invalid RST. But in reality if we encounter an invalid RST the whole program should exit so not a huge issue.